types/site.js

```diff
@@ -0,0 +1,45 @@
+ 'use strict';
+
+const Service = require('@fabric/core/types/service');
+const SPA = require('../types/spa');
+
+/**
+ * Implements a full-capacity (Native + Edge nodes) for a Fabric Site. 
+ */
+class Site extends Service {
+  /**
+   * Creates an instance of the {@link Site}, which provides general statistics covering a target Fabric node.
+   * @param {Object} [settings] Configuration values for the {@link Site}.
+   * @returns {Site} Instance of the {@link Site}.  Call `render(state)` to derive a new DOM element.
+   */
+  constructor (settings = {}) {
+    // Adopt Fabric semantics
+    super(settings);
+
+    // Define local settings
+    this.settings = Object.assign({
+      authority: 'http://localhost:9332/services/fabric', // loopback service
+      spa: null
+    }, this.settings, settings);
+
+    // Set local state
+    this._state = {
+      content: {},
+      status: 'PAUSED'
+    };
+
+    this.spa = new SPA(this.settings.spa);
+
+    // Ensure chainability
+    return this;
```

Comment:
Constructors would automatically return `this` by default, but if a return is supplied then it would follow one of these two:
```
If return is called with an object, then the object is returned instead of this.
If return is called with a primitive, it’s ignored.
```
https://javascript.info/constructor-new#return-from-constructors